pkg/apis/harvesterhci.io/v1beta1/image.go

```diff
@@ -62,6 +62,18 @@ type VirtualMachineImageSpec struct {
 	// +kubebuilder:validation:Maximum:=10
 	// +kubebuilder:validation:Optional
 	Retry int `json:"retry" default:"3"`
+
+	// +optional
+	SecurityParameters *VirtualMachineImageSecurityParameters `json:"securityParameters,omitempty"`
+}
+
+type VirtualMachineImageSecurityParameters struct {
```

Comment:
should all the fields not be mandatory when a user devices to define securityParamters.

this will ensure an additional validation by the apiserver as well

Reply:
But, those three parameters are required when users define securityParameters. I was going to prevent users from defining `securityParameters: {}` in spec. Are there any concerns about making these parameters mandatory?

```yaml
# skip
        # +optional
        SecurityParameters *VirtualMachineImageSecurityParameters `json:"securityParameters,omitempty"`
# skip

type VirtualMachineImageSecurityParameters struct {
        // +kubebuilder:validation:Required
        // +kubebuilder:validation:Enum=encrypt;decrypt;qwe
        CryptoOperation VirtualMachineImageCryptoOperationType `json:"cryptoOperation"`

        // +kubebuilder:validation:Required
        SourceImageName string `json:"sourceImageName"`

        // +kubebuilder:validation:Required
        SourceImageNamespace string `json:"sourceImageNamespace"`
}
```

Reply:
NOTE: After discussion with @ibrokethecloud, we decide to make those three required fields, and use pointer to prevent users from inputing `securityParameters: {}`.